client/src/components/BlogCard/index.jsx

```diff
@@ -1,17 +1,17 @@
 /* eslint-disable jsx-a11y/anchor-is-valid */
 import React from 'react';
 import PropTypes from 'prop-types';
+import { Link } from 'react-router-dom';
 import BlogDetails from '../BlogDetails';
 import styles from './BlogCard.module.css';
 
 export default function BlogCard({ data }) {
   const {
-    category, title, content, author, image,
+    category, title, content, author, image, _id,
   } = data;
-
   return (
     <div className={styles.container}>
-      <a className={styles.cardLink} href="">
+      <Link className={styles.cardLink} to={`/post/${_id}`}>
```

Comment:
Don't use link here, use a tag with href as the path

Reply:
react router dom handles browser's back button action better as compared to while using the same with an anchor